tutorials/image/cifar10_estimator/cifar10_main.py

```diff
@@ -460,86 +459,196 @@ def _experiment_fn(run_config, hparams):
+      allow_soft_placement=True,
+      log_device_placement=log_device_placement,
+      intra_op_parallelism_threads=num_intra_threads,
+      gpu_options=tf.GPUOptions(
+          force_gpu_compatible=force_gpu_compatible
+      )
+  )
+
+  config = tf.contrib.learn.RunConfig(
+      session_config=sess_config,
+      model_dir=job_dir)
+  tf.contrib.learn.learn_runner.run(
+      get_experiment_fn(
+          data_dir,
+          num_gpus,
+          avg_on_gpu,
+          use_distortion_for_training
+      ),
+      run_config=config,
+      hparams=tf.contrib.training.HParams(**hparams)
+  )
+
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--data-dir',
+      type=str,
+      required=True,
+      help='The directory where the CIFAR-10 input data is stored.'
+  )
+  parser.add_argument(
+      '--job-dir',
+      type=str,
+      required=True,
+      help='The directory where the model will be stored.'
+  )
+  parser.add_argument(
+      '--avg-on-gpu',
+      action='store_true',
+      default=False,
+      help='If present, use GPU to average gradients.'
+  )
+  parser.add_argument(
+      '--num-gpus',
+      type=int,
+      default=1,
+      help='The number of gpus used. Uses only CPU if set to 0.'
+  )
+  parser.add_argument(
+      '--num-layers',
+      type=int,
+      default=44,
+      help='The number of layers of the model.'
+  )
+  parser.add_argument(
+      '--train-steps',
+      type=int,
+      default=80000,
+      help='The number of steps to use for training.'
+  )
+  parser.add_argument(
+      '--train-batch-size',
+      type=int,
+      default=128,
+      help='Batch size for training.'
+  )
+  parser.add_argument(
+      '--eval-batch-size',
+      type=int,
+      default=100,
+      help='Batch size for validation.'
+  )
+  parser.add_argument(
+      '--momentum',
+      type=float,
+      default=0.9,
+      help='Momentum for MomentumOptimizer.'
+  )
+  parser.add_argument(
+      '--weight-decay',
+      type=float,
+      default=2e-4,
+      help='Weight decay for convolutions.'
+  )
+  
+  parser.add_argument(
+      '--learning-rate',
+      type=float,
+      default=0.1,
+      help="""\
+      This is the inital learning rate value. The learning rate will decrease
+      during training. For more details check the model_fn implementation in
+      this file.\
+      """
+  )
+  parser.add_argument(
+      '--use-distortion-for-training',
+      type=bool,
+      default=True,
+      help='If doing image distortion for training.'
+  )
+  parser.add_argument(
+      '--sync',
+      action='store_true',
+      default=False,
+      help="""\
+      If present when running in a distributed environment will run on sync mode.\
+      """
+  )
+  parser.add_argument(
+      '--num-intra-threads',
+      type=int,
+      default=1,
+      help="""\
+      Number of threads to use for intra-op parallelism. If set to 0, the
```

Comment:
Looks fine to me:

```
usage: cifar10_main.py [-h] --data-dir DATA_DIR --job-dir JOB_DIR
                       [--avg-on-gpu] [--num-gpus NUM_GPUS]
                       [--num-layers NUM_LAYERS] [--train-steps TRAIN_STEPS]
                       [--train-batch-size TRAIN_BATCH_SIZE]
                       [--eval-batch-size EVAL_BATCH_SIZE]
                       [--momentum MOMENTUM] [--weight-decay WEIGHT_DECAY]
                       [--learning-rate LEARNING_RATE]
                       [--use-distortion-for-training USE_DISTORTION_FOR_TRAINING]
                       [--sync] [--num-intra-threads NUM_INTRA_THREADS]
                       [--num-inter-threads NUM_INTER_THREADS]
                       [--force-gpu-compatible] [--log-device-placement]
                       [--batch_norm_decay BATCH_NORM_DECAY]
                       [--batch_norm_epsilon BATCH_NORM_EPSILON]

optional arguments:
  -h, --help            show this help message and exit
  --data-dir DATA_DIR   The directory where the CIFAR-10 input data is stored.
  --job-dir JOB_DIR     The directory where the model will be stored.
  --avg-on-gpu          If present, use GPU to average gradients.
  --num-gpus NUM_GPUS   The number of gpus used. Uses only CPU if set to 0.
  --num-layers NUM_LAYERS
                        The number of layers of the model.
  --train-steps TRAIN_STEPS
                        The number of steps to use for training.
  --train-batch-size TRAIN_BATCH_SIZE
                        Batch size for training.
  --eval-batch-size EVAL_BATCH_SIZE
                        Batch size for validation.
  --momentum MOMENTUM   Momentum for MomentumOptimizer.
  --weight-decay WEIGHT_DECAY
                        Weight decay for convolutions.
  --learning-rate LEARNING_RATE
                        This is the inital learning rate value. The learning
                        rate will decrease during training. For more details
                        check the model_fn implementation in this file.
  --use-distortion-for-training USE_DISTORTION_FOR_TRAINING
                        If doing image distortion for training.
  --sync                If present when running in a distributed environment
                        will run on sync mode.
  --num-intra-threads NUM_INTRA_THREADS
                        Number of threads to use for intra-op parallelism. If
                        set to 0, the system will pick an appropriate number.
                        The default is 1 since in this example CPU only
                        handles the input pipeline and gradient aggregation
                        (when --is-cpu-ps). Ops that could potentially benefit
                        from intra-op parallelism are scheduled to run on
                        GPUs.
  --num-inter-threads NUM_INTER_THREADS
                        Number of threads to use for inter-op parallelism. If
                        set to 0, the system will pick an appropriate number.
  --force-gpu-compatible
                        Whether to enable force_gpu_compatible in GPU_Options.
                        Check tensorflow/core/protobuf/config.proto#L69 for
                        details.
  --log-device-placement
                        Whether to log device placement.
  --batch_norm_decay BATCH_NORM_DECAY
                        Decay for batch norm.
  --batch_norm_epsilon BATCH_NORM_EPSILON
                        Epsilon for batch norm.
```